src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -114,14 +114,18 @@ public boolean isSuccess() {
     return isSuccess;
   }
 
-  private void checkRetryable() {
-    Gson gson = new Gson();
-    JsonObject jo = gson.fromJson(this.responseBody, JsonObject.class);
-    if (jo.get(ServiceNowConstants.STATUS) != null &&
-      jo.get(ServiceNowConstants.STATUS).getAsString().equals(ServiceNowConstants.FAILURE) &&
-      jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString()
-        .contains(ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED)) {
-      isRetryable = true;
+  private void checkRetryable() throws IOException {
+    try {
+      Gson gson = new Gson();
+      JsonObject jo = gson.fromJson(this.responseBody, JsonObject.class);
+      if (jo.get(ServiceNowConstants.STATUS) != null &&
+        jo.get(ServiceNowConstants.STATUS).getAsString().equals(ServiceNowConstants.FAILURE) &&
+        jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString()
+          .contains(ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED)) {
+        isRetryable = true;
+      }
+    } catch (Exception e) {
```

Comment:
I am wondering why would this gives a general exception? You are not making an IO call from here. All you are doing is parsing the response body. It should ideally be a parsing error.

Reply:
That makes sense. Catching exception a layer above.